.github/workflows/workflow.yml

```diff
@@ -42,7 +42,7 @@ jobs:
       run: |
         make test-with-coverage
     - name: Upload code coverage
-      uses: codecov/codecov-action@v4
+      uses: codecov/codecov-action@v3
```

Comment:
just curious: why are you downgrading this action? is it because of the number of failures or report uploads?

Reply:
yes, we just want to give it a try if it helps

Reply:
yeah, makes sense, I feel like the error rate significantly increased since we upgraded this action :/ 
it may help if we open PR from this upstream repo instead of our forks, though (in which case the action runner may have access to the token)

Reply:
there is actually another option we could try - use `pull_request_target` instead of `pull_request:` - in this way the token will be available also for the pull-request builds. 
@fbm3307 could you please update the PR so it uses `on.pull_request_target` instead - eg:
```
on:
  push:
    branches:
    - master
    tags-ignore:
    - '*.*'
  pull_request_target:
    types: [ opened, synchronize, reopened, ready_for_review ]
    branches:
    - master
```
and keep the v4 version?